src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Samples.cshtml

```diff
@@ -0,0 +1,292 @@
+                                        });
+
+                                        string revTitle = revision.RevisionTitle == null ? "Rev " + revision.RevisionNum : "Rev " + revision.RevisionNum + " - " + revision.RevisionTitle;
+
+                                        if (revision.SampleId == Model.Sample.SampleId)
+                                        {
+                                            <option selected value="@urlValue" data-content="@revTitle"></option>
+                                        }
+                                        else
+                                        {
+                                            <option value="@urlValue" data-content="@revTitle"></option>
+                                        }
+                                    }
+                                </select>
+                            </div>
+                        </div>
+                    }
+                    else 
+                    {
+                        @* Case 2 - There are no files on the server. *@
+                        <p> This could be due to either the blob being misconfigured or nonpresent, or by the file not existing on the server. </p>
+                        <p> Please ensure the blob is set up, and if you believe it to be the latter issue (file mistakenly removed), you may try to add a sample.</p>
+                        <p class="text-danger"> This <strong>will</strong> return a server error if the blob is not configured and present.</p>
+                        <button type="button" class="btn btn-warning" data-toggle="modal" data-target="#uploadModel">Add Usage Sample</button>
+                    }
+            </div>
+        </div>
+    </div>
+}
+else
+{
+    <div class="container-fluid">
+        <div class="mx-5 mt-2">
+            <div class="row">
+                <div class="col p-2" data-review-id=@Model.Review.ReviewId>
+                    @if (Model.SampleContent.Length == 0) 
+                    {
+                        <div class="text-muted">
+                            <p> There is no sample present in the review. </p>
+                            <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#uploadModel">Add Usage Sample</button>
+                        </div>
+                    }
+                    else
+                    {
+                        <div class="usage-sample">
+                            <i> Current sample uploaded by <strong> <a username="@Model.Sample.Author"> @Model.Sample.Author </a></strong></i>
+                            <span class="float-right">
+                                <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#updateModel">Edit Usage Sample</button>
+                                <button type="button" class="btn btn-primary" data-toggle="modal" data-target="#uploadModel">Add Usage Sample</button>
```

Comment:
Should we change the term 'add' to 'replace' in the contexts where a usage sample already exists?

Reply:
It does not replace the sample, but adds a new revision

Reply:
Ah, good point. Maybe it should say "Add new revision". Generally I also think we could shorten the text on these three buttons - what do you think of "Edit Sample", "Add new revision", and "Delete Sample"?

Reply:
I do agree they're a bit big, but those changes sound good to me. They're fairly easy to edit any time